providers/buffered/provider.go

```diff
@@ -0,0 +1,64 @@
+// © 2017 Helix OpCo LLC. All rights reserved.
+// Initial Author: jpecknerhelix
+
+package buffered
+
+import (
+	"context"
+	"io"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/myhelix/contextlogger/log"
+	"github.com/myhelix/contextlogger/providers"
+)
+
+type bufferedLogProvider struct {
+	*logrus.Entry
+}
+
+// LogProvider returns a LogProvider which outputs to writer, including the value of the "report" parameter
+// in methods which have it.
+func LogProvider(writer io.Writer) providers.LogProvider {
+	return bufferedLogProvider{
+		logrus.NewEntry(&logrus.Logger{
+			Out: writer,
+			Formatter: &logrus.TextFormatter{
+				DisableColors:   true,
+				TimestampFormat: "sometime", // Omit timestamp to make output predictable
```

Comment:
Since this is specifically a thing you want during testing, maybe call this TestableLogProvider or something? Name should make it clear it's not for production use.